pimd/pim_pim.c

```diff
@@ -124,6 +127,9 @@ void pim_sock_delete(struct interface *ifp, const char *delete_message)
 	pim_neighbor_delete_all(ifp, delete_message);
 
 	sock_close(ifp);
+
```

Comment:
It would be really good if the below code is moved to Line no. 117.
IMO, we should clean off the group information in our side first then send hello with 0 hold time and clean up the neighbor information.